ci-operator/step-registry/hypershift/mce/power/create/nodepool/hypershift-mce-power-create-nodepool-commands.sh

```diff
@@ -70,181 +65,38 @@ export IBMCLOUD_TRACE=true
-
-      [[registry.mirror]]
-        location = "brew.registry.redhat.io"
-        insecure = false
-
-    [[registry]]
-      location = "registry.redhat.io"
-      insecure = false
-      blocked = false
-      mirror-by-digest-only = true
-      prefix = ""
-
-      [[registry.mirror]]
-        location = "brew.registry.redhat.io"
-        insecure = false
-
-    [[registry]]
-      location = "registry.redhat.io/rhacm2"
-      insecure = false
-      blocked = false
-      mirror-by-digest-only = true
-      prefix = ""
-
-      [[registry.mirror]]
-        location = "brew.registry.redhat.io/rhacm2"
-        insecure = false
-
-    [[registry]]
-      location = "registry-proxy.engineering.redhat.com"
-      insecure = false
-      blocked = false
-      mirror-by-digest-only = true
-      prefix = ""
-
-      [[registry.mirror]]
-        location = "brew.registry.redhat.io"
-        insecure = false
-
-    [[registry]]
-      location = "registry.redhat.io/multicluster-engine"
-      insecure = false
-      blocked = false
-      mirror-by-digest-only = true
-      prefix = ""
-
-      [[registry.mirror]]
-        location = "brew.registry.redhat.io/multicluster-engine"
-        insecure = false
-EOF
-
+MCE_VERSION=$(oc get "$(oc get multiclusterengines -oname)" -ojsonpath="{.status.currentVersion}" | cut -c 1-3)
+ARCH="ppc64le"
+if (( $(echo "$MCE_VERSION < 2.6" | bc -l) )); then
+ # Support for using ppc64le arch added after MCE 2.6
+ ARCH="amd64"
+fi
 
-# Creating AgentServiceConfig
-CLUSTER_VERSION=$(oc get clusterversion -o jsonpath={..desired.version} | cut -d '.' -f 1,2)
-OS_IMAGES=$(jq --arg CLUSTER_VERSION "${CLUSTER_VERSION}" '[.[] | select(.openshift_version == $CLUSTER_VERSION)]' "${SHARED_DIR}/default_os_images.json")
-echo "$(date) Creating AgentServiceConfig"
+echo "$(date) Creating Nodepool"
 cat <<EOF | oc create -f -
-apiVersion: agent-install.openshift.io/v1beta1
-kind: AgentServiceConfig
+apiVersion: hypershift.openshift.io/v1beta1
+kind: NodePool
 metadata:
-  name: agent
+  name: ${HOSTED_CLUSTER_NAME}-ppc64le
+  namespace: ${CLUSTERS_NAMESPACE}
 spec:
-  databaseStorage:
-    accessModes:
-    - ReadWriteOnce
-    resources:
-      requests:
-        storage: ${DB_VOLUME_SIZE}
-  filesystemStorage:
-    accessModes:
-    - ReadWriteOnce
-    resources:
-      requests:
-        storage: ${FS_VOLUME_SIZE}
-  mirrorRegistryRef:
-    name: mirror-config
-  osImages:
-    - openshiftVersion: "${CLUSTER_VERSION}"
-      version: $(echo "$OS_IMAGES" | jq -r '.[] | select(.cpu_architecture == "ppc64le").version')
-      url: $(echo "$OS_IMAGES" | jq -r '.[] | select(.cpu_architecture == "ppc64le").url')
-      cpuArchitecture: ppc64le
+  arch: ${ARCH}
+  clusterName: ${HOSTED_CLUSTER_NAME}
+  management:
+    autoRepair: false
+    upgradeType: InPlace
+  nodeDrainTimeout: 0s
+  platform:
+    agent:
+      agentLabelSelector:
+        matchLabels:
+          inventory.agent-install.openshift.io/cpu-architecture: ppc64le
```

Comment:
Is this code valid when `ARCH="amd64"`?

Reply:
`hypershift-mce-power-create-nodepool` step is designed specifically for power, so directly used `ppc64le` arch. We are planning to create a separate step to handle the things for amd64 arch like `hypershift-mce-heterogeneous-nodepool-amd64-create-nodepool`(name notation is still not fixed, will add it in further PRs)

Reply:
Yes it will work for amd64 as well, but the value used is `x86_64` not `amd64`.